src/hotspot/share/cds/cdsConfig.cpp

```diff
@@ -23,10 +23,15 @@
  */
 
 #include "precompiled.hpp"
+#include "cds/archiveHeapLoader.hpp"
 #include "cds/cdsConfig.hpp"
 #include "cds/heapShared.hpp"
+#include "classfile/classLoaderDataShared.hpp"
+#include "logging/log.hpp"
 
 bool CDSConfig::_is_dumping_dynamic_archive = false;
+bool CDSConfig::_enable_dumping_full_module_graph = true;
+bool CDSConfig::_enable_loading_full_module_graph = true;
```

Comment:
The "enable" naming isn't quite right - these fields hold the enabled state e.g. `_dumping_full_module_graph_enabled`.

Reply:
I like the variable name to be the same as the function name, so it's easier to search. We also have `bool _enable_preview;` in arguments.hpp

Reply:
But the context is different. `enable_xxx()` indicates that the function enables xxx. The variable that holds the enabled/disabled state of xxx should reflect in its name that it holds that state so `_xxx_enabled` Or `_has_xxx`, or `_is_xxx`, depending on what `xxx` actually is.

Reply:
Or `_use_xxx` as was originally used here. I guess I just find it particularly irksome that `enable` is often used where `enabled` would be the grammatically correct choice.

Reply:
I don't think enable_xxx is grammatically wrong, when it's read as "(we should) enable xxx". We have hundreds of global flags that start with Use/Enable/Disable, like EnableDynamicAgentLoading and UseTLAB.

Anyway, the reason I didn't use `_is_loading_full_module_graph` is that I want a flag that unconditionally disables the feature, but this flag alone doesn't guarantee that the feature exists.

I changed the flag to `_loading_full_module_graph_disabled` and added comments to make it clear.